server/src/com/cloud/resource/ResourceManagerImpl.java

```diff
@@ -1148,15 +1148,16 @@ public Host cancelMaintenance(final CancelMaintenanceCmd cmd) {
     }
 
     @Override
-    public Host reconnectHost(final ReconnectHostCmd cmd) {
-        final Long hostId = cmd.getId();
+    public Host reconnectHost(ReconnectHostCmd cmd) throws CloudRuntimeException, AgentUnavailableException{
```

Comment:
Does this method need to return a host? It seems that it just reconnects to a host, if not, throws an exception.
If so, it could be void as well.